### ISSUE TITLE
Fix file paths that contain the `+` character being ignored

### ISSUE THREAD

**Issue by andrewthauer** [stylelint/stylelint#4211]
> Clearly describe the bug

File paths that contain a `+` are either not linted or will fail with an error.

If the glob pattern itself contains a `+` itself it will throw this error even if it's quoted or escaped:

```
Error: No files matching the pattern "src/app/+route" were found.
    at globby.then.filePaths (/.../node_modules/stylelint/lib/standalone.js:206:17)
    at process._tickCallback (internal/process/next_tick.js:68:7)
error Command failed with exit code 1.
```

If the glob pattern does not contain a `+` but the pattern would match paths that do contain a `+` it will not error, but it also won't lint those files either. It seems to just ignore them.

> Which rule, if any, is the bug related to?

n/a

> What CSS is needed to reproduce the bug?

n/a

> What stylelint configuration is needed to reproduce the bug?

n/a

> Which version of stylelint are you using?

`10.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

- `stylelint src/app/+route/styles.scss`
- `stylelint src/app/+route`
- `stylelint 'src/app/+route/styles.scss'`
- `stylelint src/app/\+route/styles.scss`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

n/a

> What did you expect to happen?

It will lint file paths that contain the `+` symbol it does not throw an error and also lints files that might be in paths that contain a `+`.

> What actually happened (e.g. what warnings or errors did you get)?

Doesn't lint the files or throws an error.

Some initial investigation seems to suggest this might be an issue in `globby`. I tried updating `globby` to the latest version and it no longer threw the errors. However, it also didn't pick up actual linting errors for any files contained with paths that contain `+`. I've tried the same with other linters `TSLint` & `ESLint` and both seem to handle these paths fine. However, they also don't use `globby` either.

**Comment by evilebottnawi** [comment]
I.e. new version globby is work as expected?

**Comment by andrewthauer** [comment]
I tried switching to `globby@10.0.1` (latest version) and it still doesn't lint file paths containing `+`. It essentially just ignores any file paths that have a `+` in it somewhere. I can't be sure it's `globby` but feels like a possibility. I'll try and find some time to see if I can reproduce the issue with globby alone.

**Comment by jeddy3** [comment]
@andrewthauer Thanks for digging deeper. I'll mark up as a bug needing help.

**Comment by hudochenkov** [comment]
It looks like it should be fixed in the latest `13.7.2` release. @andrewthauer, could you check, please?
